src/features/HelpAndKnowledge/SolutionsHelp/SolutionDetails/Solutions/Generic/_components/ContractorForm/index.tsx

```diff
@@ -56,24 +65,45 @@ const ContractorForm = ({
   const {
     control,
     handleSubmit,
-    reset,
     formState: { isSubmitting, isDirty, isValid }
   } = methods;
 
-  const { selectedSolution } = useModalSolutionState();
-  const { showMessage } = useMessage();
-  const [create] = useCreateMtoCommonSolutionContractorMutation();
-  const [update] = useUpdateMtoCommonSolutionContractorMutation();
+  const [create] = useCreateMtoCommonSolutionContractorMutation({
+    refetchQueries: [
+      {
+        query: GetMTOSolutionContacts
+      }
+    ]
+  });
+
+  const [update] = useUpdateMtoCommonSolutionContractorMutation({
+    refetchQueries: [
+      {
+        query: GetMTOSolutionContacts
+      }
+    ]
+  });
+
   const [mutationError, setMutationError] = useState<
     'duplicate' | 'generic' | null
   >(null);
+
   const disabledSubmitBtn = isSubmitting || !isDirty || !isValid;
 
+  useEffect(() => {
+    setDisableButton(disabledSubmitBtn);
+  }, [setDisableButton, disabledSubmitBtn]);
+
+  useEffect(() => {
+    setSubmitForm(() => onSubmit);
```

Comment:
I noticed that all the mutations are firing twice, with the first one failing.  I think the culprit is where and how you are setting the onsubmit handler to the parent.  Additionally in the parent you will need to change how you are passing the setter, I'll note that in another area

```suggestion
    setSubmitForm(onSubmit);
```